nikola/plugins/command/check.py

```diff
@@ -44,33 +45,23 @@
 from nikola.utils import get_logger, req_missing
 
 
-def _call_nikola_list(l, site, arguments):
-    class NotReallyAStream(object):
-        """A massive hack."""
-        out = []
+def _call_nikola_list(site):
+    files = []
+    for task in generate_tasks('render_site', site.gen_tasks('render_site', "Task", '')):
+        files.extend(task.targets)
+    for task in generate_tasks('post_render', site.gen_tasks('render_site', "LateTask", '')):
+        files.extend(task.targets)
+    return files
 
-        def write(self, t):
-            self.out.append(t)
 
-    oldstream = l.outstream
-    newstream = NotReallyAStream()
-    try:
-        l.outstream = newstream
-        l.parse_execute(arguments)
-        return newstream.out
-    finally:
-        l.outstream = oldstream
-
-
-def real_scan_files(l, site):
+def real_scan_files(site):
     task_fnames = set([])
     real_fnames = set([])
     output_folder = site.config['OUTPUT_FOLDER']
     # First check that all targets are generated in the right places
-    for task in _call_nikola_list(l, site, ["--all"]):
-        task = task.strip()
-        if output_folder in task and ':' in task:
-            fname = task.split(':', 1)[-1]
+    for fname in _call_nikola_list(site):
+        fname = fname.strip()
+        if output_folder in fname:
```

Comment:
`startwith()` is better than `in` in this context, as maybe some file `cache/bla/output/bla` is generated, which also contains `output`.